modules/geo-layers/src/quadkey-layer/quadkey-utils.js

```diff
@@ -0,0 +1,28 @@
+import {worldToLngLat} from '@math.gl/web-mercator';
+
+const TILE_SIZE = 512;
+
+export function quadkeyToWorldBounds(quadkey) {
+  let x = 0;
+  let y = 0;
+  let mask = 1 << quadkey.length;
+  const scale = mask / TILE_SIZE;
+
+  for (let i = 0; i < quadkey.length; i++) {
+    mask >>= 1;
+    const q = parseInt(quadkey[i]);
+    if (q % 2) x |= mask;
+    if (q > 1) y |= mask;
+  }
+  return [
+    [x / scale, TILE_SIZE - y / scale],
+    [(x + 1) / scale, TILE_SIZE - (y + 1) / scale]
+  ];
+}
+
+export function getQuadkeyPolygon(quadkey) {
+  const [topLeft, bottomRight] = quadkeyToWorldBounds(quadkey);
+  const [w, n] = worldToLngLat(topLeft);
+  const [e, s] = worldToLngLat(bottomRight);
+  return new Float64Array([e, n, e, s, w, s, w, n, e, n]);
```

Comment:
It is not necessary to convert the coordinates to a typed array. It only adds performance overheads because of extra GC.

Reply:
Fixed. Is there a reason why a `Float64Array` is created in `s2-utils.js`?

https://github.com/visgl/deck.gl/blob/master/modules/geo-layers/src/s2-layer/s2-utils.js#L42